plugins/flatpak/gs-flatpak.c

```diff
@@ -3679,72 +3680,73 @@ app_has_local_source (GsApp *app)
 }
 
 static guint64
-get_installation_dir_free_space (GsFlatpak *self)
+get_installation_dir_free_space (GsFlatpak *self, GError **error)
 {
 	g_autoptr (GFile) installation_dir = NULL;
 	g_autoptr (GFileInfo) info = NULL;
-	g_autoptr (GError) local_error = NULL;
-	guint64 free_space = 0;
 
 	installation_dir = flatpak_installation_get_path (self->installation);
 
 	info = g_file_query_filesystem_info (installation_dir,
 					     G_FILE_ATTRIBUTE_FILESYSTEM_FREE,
 					     NULL,
-					     &local_error);
+					     error);
 	if (info == NULL)
-		g_warning ("Failed to query filesystem info: %s", local_error->message);
+		return 0;
 
-	free_space = g_file_info_get_attribute_uint64 (info, G_FILE_ATTRIBUTE_FILESYSTEM_FREE);
-	return free_space;
+	return g_file_info_get_attribute_uint64 (info, G_FILE_ATTRIBUTE_FILESYSTEM_FREE);
 }
 
 static gboolean
 gs_flatpak_has_space_to_install (GsFlatpak *self, GsApp *app, GsAppList *list)
 {
+	g_autoptr(GError) error = NULL;
 	guint64 free_space = 0;
 	guint64 space_required = 0;
 
 	space_required = gs_app_get_size_download (app);
 	if (space_required == GS_APP_SIZE_UNKNOWABLE)
 		g_warning ("Failed to query download size: %s", gs_app_get_unique_id (app));
 
-	space_required = space_required + (500 * 1024 * 1024); /* min-free-space-size = 500MB for Endless */
-	free_space = get_installation_dir_free_space (self);
-
-	if (free_space < space_required)
-		return FALSE;
+	space_required = space_required + EOS_MIN_FREE_SPACE_NEEDED_BYTES;
+	free_space = get_installation_dir_free_space (self, &error);
+	if (error != NULL) {
+		g_warning ("Error installing: %s", error->message);
```

Comment:
It was not an error installing, it was an error getting the free space. And you should always mention which app it happened for, otherwise the message will not be very useful.
`Error getting the free space available for installing %s: %s` -> app unique id , error msg